provider/provider_yaml_test.go

```diff
@@ -209,6 +209,24 @@ func TestNoGlobalProjectWarning(t *testing.T) {
 	)
 }
 
+func TestGlobalProjectNoProjectWarning(t *testing.T) {
+	if testing.Short() {
+		t.Skipf("Skipping in testing.Short() mode, assuming this is a CI run without GCP creds")
+	}
+	cwd, err := os.Getwd()
+	require.NoError(t, err)
+
+	test := pulumitest.NewPulumiTest(t, "test-programs/noproject-bucket",
+		opttest.LocalProviderPath(providerName, filepath.Join(cwd, "..", "bin")))
+
+	test.SetConfig(t, "gcpProj", "No Project")
+	res := test.Up(t)
```

Comment:
Do we have sufficient confidence that the added "No Project" only disables spurious warnings?
Asked another way - do we need to add a test for the situation (which I assume is the original trigger for #1168) where the global setting is "No Project" and the project is set inside provider config?

Reply:
I added in this test, `provider/test-programs/noproject-bucket/Pulumi.yaml` but I can't run it locally because I had to hardcode the gcp project, and I am getting, `richard@pulumi.com does not have storage.buckets.create access to the Google Cloud project` which makes sense. I opened the PR to let the test bootstrap run it. I couldn't think of a good way around hardcoding the project, but would love suggestions on that one, so I can test locally.

Reply:
@iwahbe @guineveresaenger do we want to add a project to this test: https://www.pulumi.com/registry/packages/gcp/api-docs/organizations/project/, I think this might match the ticket more, but I worry about leaking the orgId?

Reply:
If this runs in CI, I'm fine using `pulumi-developtment`.

Reply:
I don't think we need to test with the project resource if thats what you're asking.